shared/src/iosMain/kotlin/com/karumi/gallery/domain/TimeProvider.kt

```diff
@@ -0,0 +1,36 @@
+package com.karumi.gallery.domain
+import kotlinx.cinterop.alloc
+import kotlinx.cinterop.memScoped
+import kotlinx.cinterop.ptr
+import platform.Foundation.NSUserDefaults
+import platform.posix.gettimeofday
+import platform.posix.timeval
+
+actual class TimeProvider {
+
+  companion object {
+    private const val TIME_ARG = "time_arg"
+    private const val THREE_MONTHS_IN_MILLIS = 259200
+  }
+
+  private val delegate: NSUserDefaults = NSUserDefaults.standardUserDefaults
+
+  actual fun persistTime() {
+    val timestampMillis = getCurrentTime()
+    delegate.setInteger(timestampMillis, TIME_ARG)
+  }
+
+  actual fun getPersistedTime(): Long = delegate.integerForKey(TIME_ARG).let {
+    if (it.equals(0)) timeAgo() else it
+  }
+
+  actual fun getCurrentTime(): Long = memScoped {
+    val timeVal = alloc<timeval>()
+    gettimeofday(timeVal.ptr, null)
+    val sec = timeVal.tv_sec
+    val usec = timeVal.tv_usec
+    ((sec * 1_000L) + (usec / 1_000L))
```

Comment:
Fancy 🤓

Reply:
I spent too much time to find how to get a long timestamp on swift, it returns Double -.- with `NSDate.timeIntervalSince1970`